commons-db/src/main/java/com/palantir/nexus/db/sql/SQLString.java

```diff
@@ -246,27 +232,29 @@ private SQLString(String sql) {
      * @param dbType This is only used in making the SQL comment
      */
     protected SQLString(String key, String sql, DBType dbType) {
-        super(key, makeCommentString(key, dbType) + sql);
+        super(key, prependPrefix(key, dbType, sql));
     }
 
     /**
-     * Creates an appropriate comment string for the beginning of a SQL statement.
-     *
+     * Creates a SQL string prepended with a comment indicating the registration and DB type.
      * @param keyString Identifier for the SQL; will be null if the SQL is unregistered
      * @param dbType    The database type
+     * @param sql       The SQL to append
+     * @return string with comment prepended
      */
-    private static String makeCommentString(String keyString, DBType dbType) {
-        String registrationState;
-        if (keyString != null) {
-            registrationState = "SQLString Identifier: " + keyString; // $NON-NLS-1$
-        } else {
-            registrationState = "UnregisteredSQLString"; // $NON-NLS-1$
-        }
-        String dbTypeString = ""; // $NON-NLS-1$
-        if (dbType != null) {
-            dbTypeString = " dbType: " + dbType; // $NON-NLS-1$
+    @VisibleForTesting
+    static String prependPrefix(String keyString, DBType dbType, String sql) {
+        if (keyString == null && dbType == null) {
+            return unregisteredSql(sql);
         }
-        return "/* " + registrationState + dbTypeString + " */ "; // $NON-NLS-1$ //$NON-NLS-2$
+        return "/* " // $NON-NLS-1$
+                + (keyString == null ? UNREGISTERED_SQL_STRING : "SQLString Identifier: " + keyString) // $NON-NLS-1$
+                + (dbType == null ? "" : " dbType: " + dbType) // $NON-NLS-1$
+                + " */ " + sql; // $NON-NLS-1$
```

Comment:
This should now be optimized string concatenation via [JEP 280](https://openjdk.org/jeps/280)